### DISCUSSION TITLE
Import Statements Question

### DISCUSSION THREAD

**Discussion by max-muoto** [microsoft/pyright#6891] [General]
From this doc: https://github.com/microsoft/pyright/blob/main/docs/import-statements.md

I was trying to understand if it's intentional this isn't allowed:

```python
from exteral_module.module_b import Class1, module_c
import module_b
```

Where even if you're using a `from` statement you can't have your own module named the same as the external module's sub module. I was doing something like this and ran into `is not a known member of module` errors.

**Comment by erictraut** [comment]
I don't think I fully understand your question. Could you describe the layout of the modules and submodules and explain whether they are traditional or namespace packages (i.e. whether an `__init__.py` is present)?

**Reply by max-muoto** [reply]
The layout is effectively:

```
Project/
  pyrightconfig.json
  src/ # No `__init__.py` in `src`
    module_a
        __init__.py
        main.py
    module_b
        __init__.py
        main.py
```

It turns out we had another duplicate top-level folder (in Project) not being used as a Python module, which was causing this.

**Reply by erictraut** [reply]
OK, I'm still not sure I fully understand the scenario. Sounds like it had nothing to do with an external module. I'm glad to hear you figured out source of the problem.
